crates/astria-sequencer/src/accounts/transaction.rs

```diff
@@ -0,0 +1,72 @@
+use anyhow::{
+    ensure,
+    Result,
+};
+use async_trait::async_trait;
+use serde::{
+    Deserialize,
+    Serialize,
+};
+
+use crate::{
+    accounts::{
+        state_ext::{
+            StateReadExt,
+            StateWriteExt,
+        },
+        types::{
+            Address,
+            Balance,
+            Nonce,
+        },
+    },
+    transaction::ActionHandler,
+};
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+pub struct Transaction {
+    to: Address,
+    from: Address,
+    amount: Balance,
+    nonce: Nonce,
+}
+
+impl Transaction {
+    pub fn new(to: Address, from: Address, amount: Balance, nonce: Nonce) -> Self {
+        Self {
+            to,
+            from,
+            amount,
+            nonce,
+        }
+    }
+}
+
+#[async_trait]
+impl ActionHandler for Transaction {
+    fn check_stateless(&self) -> Result<()> {
+        Ok(())
+    }
+
+    async fn check_stateful<S: StateReadExt + 'static>(&self, state: &S) -> Result<()> {
+        let curr_nonce = state.get_account_nonce(&self.from).await?;
+
+        // TODO: do nonces start at 0 or 1? this assumes an account's first tx has nonce 1.
```

Comment:
I believe each account nonce balance should start at `0` and the first transaction should be `1`

Reply:
cool, that's how it functions right now, the first tx has nonce 1